src/components/NewMap/ColorKeyControl/index.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useCallback, useState } from 'react';
+
+import { Popover, Typography, useTheme } from '@mui/material';
+import { Icon } from '@terraware/web-components';
+
+import { useMapPortalContainer } from 'src/components/Map/MapRenderUtils';
+import { useLocalization } from 'src/providers';
+import { ACTIVITY_TYPES, activityTypeColor, activityTypeLabel } from 'src/types/Activity';
+
+export type MapLayer = 'Planting Site' | 'Zones' | 'Sub-Zones' | 'Monitoring Plots' | 'Project Zones';
+
+export default function ColorKeyControl(): JSX.Element | null {
+  const theme = useTheme();
+  const mapPortalContainer = useMapPortalContainer();
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const handleClick = (event?: React.MouseEvent<HTMLButtonElement, MouseEvent> | undefined) => {
+    setAnchorEl(event?.currentTarget ?? null);
+  };
+  const { strings } = useLocalization();
+
+  const handleControlClose = useCallback(() => {
+    setAnchorEl(null);
+  }, []);
+
+  if (mapPortalContainer) {
+    return null;
+  }
+
+  return (
+    <>
+      <Popover
+        id='color-key-popover'
+        open={Boolean(anchorEl)}
+        anchorEl={anchorEl}
+        onClose={handleControlClose}
+        anchorOrigin={{
+          vertical: 'top',
+          horizontal: 'left',
+        }}
+        transformOrigin={{
+          vertical: 'top',
+          horizontal: 'left',
+        }}
+        sx={{
+          '& .MuiPaper-root': {
+            padding: theme.spacing(1),
+            borderRadius: '8px',
+            overflow: 'visible',
+          },
+        }}
+      >
+        <table>
+          <thead>
+            <tr>
+              <th>
+                <Typography fontSize={'10px'} fontWeight={600} textAlign={'left'}>
+                  {strings.COLOR}
+                </Typography>
+              </th>
+              <th>
+                <Typography fontSize={'10px'} fontWeight={600} textAlign={'left'}>
+                  {strings.KEY}
+                </Typography>
+              </th>
+            </tr>
+          </thead>
+          <tbody>
+            {ACTIVITY_TYPES.map((at) => (
+              <tr key={at}>
+                <td style={{ display: 'flex', height: '20px', alignItems: 'center' }}>
+                  <Icon name='iconPhoto' fillColor={activityTypeColor(at)} />
+                </td>
+                <td>
+                  <Typography fontSize={'12px'}>{activityTypeLabel(at, strings)}</Typography>
+                </td>
+              </tr>
+            ))}
```

Comment:
NIT: It might be nice to just use `activityType` or something longer than `at`, but not a blocker.